test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -322,15 +341,18 @@ func BrokerV1Beta1ConsumerDataPlaneTestHelper(
 			source := "filtered-event"
 			event.SetSource(source)
 			secondEvent := baseEvent
-
-			client.SendEventToAddressable("first-"+source+"-sender", broker.Name, testlib.BrokerTypeMeta, event, sender.WithEncoding(ce.EncodingStructured))
-			client.SendEventToAddressable("second-"+source+"-sender", broker.Name, testlib.BrokerTypeMeta, secondEvent, sender.WithEncoding(ce.EncodingStructured))
+			firstSenderName := "first-" + source + "-sender"
+			secondSenderName := "second-" + source + "-sender"
+			client.SendEventToAddressable(firstSenderName, broker.Name, testlib.BrokerTypeMeta, event, sender.WithEncoding(ce.EncodingStructured))
+			client.SendEventToAddressable(secondSenderName, broker.Name, testlib.BrokerTypeMeta, secondEvent, sender.WithEncoding(ce.EncodingStructured))
 			filteredEventMatcher := recordevents.MatchEvent(
 				cetest.HasSource(source),
 			)
 			nonEventMatcher := recordevents.MatchEvent(
 				cetest.HasSource(baseSource),
 			)
+			client.WaitForResourceReadyOrFail(firstSenderName, &podMeta)
```

Comment:
This certainly will make things better, but I'm still curious if there's a race here since this checks that the pod is running, and I assume it's fairly quick to send the event, but still seems like knowing that the pod is ready and it actually has sent the event are not exactly the same thing? I wonder if actually using something like a job that has successfully completed would be a better signal to key off of that the event has supposedly been successfully sent?